app/src/components/Graphs/EngagementScore/EngagementScoreChart.tsx

```diff
@@ -0,0 +1,201 @@
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    fetchEngagementStats();
+  }, [filters]);
+
+  // Calculate engagement...
+  const calculateEngagement = () => {
+    // Evaluate double-text score - want this as close to zero as possible
+    const doubleTextDiff = Math.abs(
+      getValFromResult(doubleTextResults, true) -
+        getValFromResult(doubleTextResults, false)
+    );
+
+    const doubleTextScore = 1 - doubleTextDiff / (totalSent + totalReceived);
+
+    // Evaluate left on read numbers. Ideally as low as possible, and as close to each other as possible
+    const leftonReadDiff = Math.abs(
+      getValFromResult(leftonreadResults, true) -
+        getValFromResult(leftonreadResults, false)
+    );
+
+    const leftonReadTotal =
+      getValFromResult(leftonreadResults, true) +
+      getValFromResult(leftonreadResults, false);
+
+    const leftonReadScore =
+      (1 - leftonReadDiff / (totalSent + totalReceived)) *
+      (1 - leftonReadTotal / (totalSent + totalReceived));
+
+    // Evaluate message length. Ideally as high as possible and as close to each other as possible
+    const avgLengthDiff = Math.abs(
+      getValFromResult(avgLengthResults, true) -
+        getValFromResult(avgLengthResults, false)
+    );
+    const avgLengthTotal =
+      getValFromResult(avgLengthResults, true) +
+      getValFromResult(avgLengthResults, false);
+
+    const avgLengthScore =
+      (1 - avgLengthDiff / (totalSent + totalReceived)) *
+      (avgLengthTotal / (totalSent + totalReceived));
+
+    // Evaluate average delay numbers. Ideally as low as possible, and as close to each other as possible
+    const avgDelayDiff = Math.abs(
+      getValFromResult(avgDelayResults, true) -
+        getValFromResult(avgDelayResults, false)
+    );
+
+    const avgDelayTotal =
+      getValFromResult(avgDelayResults, true) +
+      getValFromResult(avgDelayResults, false);
+
+    const avgDelayScore =
+      (1 - avgDelayDiff / (totalSent + totalReceived)) *
+      (1 - avgDelayTotal / (totalSent + totalReceived));
+
+    return (
+      2 * doubleTextScore +
+      leftonReadScore +
+      3 * avgLengthScore +
+      3 * avgDelayScore
+    );
+  };
+
+  const engagementScore = calculateEngagement();
+
+  return (
+    <GraphContainer title={title} description={description} icon={icon}>
+      <div style={{ display: 'flex', marginTop: 25 }}>
+        <Stat>
+          <div style={{ display: 'flex', alignItems: 'center' }}>
+            <div>
+              <Text
+                bgGradient="linear(to-br, #7928CA, #FF0080)"
+                bgClip="text"
+                fontSize="6xl"
+                fontWeight="extrabold"
+              >
+                {engagementScore.toLocaleString()}
+              </Text>
+            </div>
+            <div style={{ marginLeft: 16 }}>
+              <Text>
+                <StatArrow type="increase" />
+                9.05%
```

Comment:
Reminder this is hardcoded
```suggestion
                9.05%
```